R/tooltips.R

```
@@ -190,3 +192,217 @@ body(saveAndGetTooltips)[[length(body(saveAndGetTooltips))]] <- quote(
     addAttributes = addAttributes
   )
 )
+
+
+#' Tooltips and coords for piechart
+#'
+#' @param pp plot of class \code{ggplot}; must be in polar coordinates
+#' @inheritParams getTooltips
+#'
+#' @details Currently only single layer plots are handled
+#'
+#' @return
+#'
+getTooltipsForPiechart <- function(pp, varDict) {
+  if (length(pp$layers) > 1) {
+    warning("Only single layer plots are accepted")
+    return(NULL)
```

I wonder if we should "fail silently" here. What will happen here when someone runs this in their Shiny app? Will the tooltips be disabled (desired behavior)?

---

Maybe we should fail, maybe not. The app won't crash. It's not fully connected with front-end code so it's hard for me to fully comment the behavior presented here. It's rather a matter of agreement what should happen. Eventually, multilayer plots should be handled as well. The problem of tooltips for multilayer charts is not resolved at all currently and should be discussed separately.